<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="bagpath" default="jackal"/>

  <node pkg="topic_tools" name="downsample_ovc" type="throttle" args="messages /ovc/rgb/image_raw 2.0 /ovc/rgb/image_throttle"/>
  <node pkg="image_transport" name="compress_ovc" type="republish" args="raw in:=/ovc/rgb/image_throttle compressed out:=/ovc/rgb/image_comp"/>

  <node pkg="topic_tools" name="downsample_map" type="throttle" args="messages /top_down_render/map/compressed 2.0 /top_down_render/map_throttle/compressed"/>

  <node pkg="topic_tools" name="downsample_seg" type="throttle" args="messages /edgetpu_semantic_segmenter/output/image/compressed 2.0 /edgetpu_semantic_segmenter/output/image_throttle/compressed"/>

  <node pkg="rosbag" name="bag" type="record" args="-o /home/dcist/data/$(arg bagpath) 
    /tf 
    /tf_static
    /os1_node/imu_packets
    /os1_node/lidar_packets
    /ovc/rgb/image_comp/compressed
    /edgetpu_semantic_segmenter/output/image_throttle/compressed
    /upslam/global_pose
    /upslam/sem_pose
    /upslam/graph_viz
    /upslam/keyframe_images/compressedDepth
    /upslam/map_graph
    /top_down_render/map_throttle/compressed
    /map
    /move_base/global_costmap/costmap
    /move_base/local_costmap/costmap
    /move_base/GlobalPlanner/plan
    /multiwaypoint/path_viz
    /multiwaypoint/waypt_viz
    /multiwaypoint/next_target_viz
    /multiwaypoint/target_region_viz
    /robot/odom
    /robot/det_odom
  "/>

</launch>
